tests/sample/map.c

```diff
@@ -0,0 +1,169 @@
+        }                                                                                                             \
+    }
+
+#define POP_VALUE(MAP, EXPECTED_VALUE, EXPECT_RESULT)                                                \
+    {                                                                                                \
+        uint32_t new_value = 0;                                                                      \
+        int result = bpf_map_pop_elem(MAP, &new_value);                                              \
+        if ((EXPECT_RESULT) != result) {                                                             \
+            bpf_printk("bpf_map_pop_elem expecting result %d returned %d", (EXPECT_RESULT), result); \
+            return result;                                                                           \
+        } else if (new_value != (EXPECTED_VALUE)) {                                                  \
+            bpf_printk("bpf_map_pop_elem return %d expecting %d", new_value, (EXPECTED_VALUE));      \
+            return -1;                                                                               \
+        }                                                                                            \
+    }
+
+#define PEEK_VALUE(MAP, EXPECTED_VALUE, EXPECT_RESULT)                                                \
+    {                                                                                                 \
+        uint32_t new_value = 0;                                                                       \
+        int result = bpf_map_peek_elem(MAP, &new_value);                                              \
+        if ((EXPECT_RESULT) != result) {                                                              \
+            bpf_printk("bpf_map_peek_elem expecting result %d returned %d", (EXPECT_RESULT), result); \
+            return result;                                                                            \
+        } else if (new_value != (EXPECTED_VALUE)) {                                                   \
+            bpf_printk("bpf_map_peek_elem return %d expecting %d", new_value, (EXPECTED_VALUE));      \
+            return -1;                                                                                \
+        }                                                                                             \
+    }
+
+struct _ebpf_map_definition_in_file STACK_map;
+
+__attribute__((always_inline)) int
+test_PUSH_POP_map(struct _ebpf_map_definition_in_file* map)
+{
+    int i;
+    PEEK_VALUE(map, 0, -7);
+
+    for (i = 0; i < 10; i++) {
+        PUSH_VALUE(map, i, FALSE, 0);
+    }
+
+    PUSH_VALUE(map, 10, FALSE, -29);
+    PUSH_VALUE(map, 10, TRUE, 0);
+
+    PEEK_VALUE(map, (map == &STACK_map) ? 10 : 1, 0);
+
+    for (i = 0; i < 10; i++) {
+        POP_VALUE(map, (map == &STACK_map) ? 10 - i : i + 1, 0);
+    }
+
+    PEEK_VALUE(map, 0, -7);
+
```

Comment:
Also add a test for `POP_VALUE` when the stack/queue is empty, here and by 119, to verify a negative value is returned by bpf_map_pop_value() too.